### PR TITLE
Optionally expose api server in worker

### PR DESCRIPTION
**Pull request by lferran**
to be able to collect metrics

### CODE REVIEW

File: guillotina_amqp/commands/worker.py [hunk 1]
```diff
@@ -61,14 +63,28 @@ class WorkerCommand(Command):
...
                             help='Max simultaneous running tasks',
                             type=int, default=None)
+        parser.add_argument('--metrics-server', help='Launch an API server to expose metrics',
+                            default=False, action='store_true')
         return parser
 
-    async def run(self, arguments, settings, app):
+    def run(self, arguments, settings, app):
+        loop = self.get_loop()
+        fut = loop.run_in_executor(None, self.run_worker, arguments, settings, app)
```
Comment (vangheem) [inline guillotina_amqp/commands/worker.py]: I think you want to use `asyncio.ensure_future` here only and not with `run_in_executer`.

Also, I think it should be conditional and don't don this way if you are using the metrics server!

Comment (lferran) [reply guillotina_amqp/commands/worker.py]: I tried with ensure_future already, but the worker wasn't picking up new tasks. I will give it another shot :+1: 

It is actually conditional on the parameter `--metrics-server`.

**Comment by lferran** [comment]
@dmanchon @vangheem seems that when we call web.run_app(), the loop is already running and it fails.
At the same time, the worker needs the same loop than the guillotina app in order to function.

any idea how to work this around?

**Comment by lferran** [comment]
Other things I'm coming across:
1) Seems like the `SeverCommand.run` can't be ran in a separate thread, otherwise: 
```
Traceback (most recent call last):
  File "/home/ferran/.pyenv/versions/3.6.4/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/ferran/onna/guillotina_amqp/guillotina_amqp/commands/worker.py", line 29, in run
    self.arguments, self.settings, self.app)
  File "/home/ferran/onna/guillotina_amqp/guillotina_amqp/commands/worker.py", line 110, in run_server
    super().run(arguments, settings, app)
  File "/home/ferran/onna/onna-canonical/venv/lib/python3.6/site-packages/guillotina/commands/server.py", line 49, in run
    access_log_format=log_format)
  File "/home/ferran/onna/onna-canonical/venv/lib/python3.6/site-packages/aiohttp/web.py", line 60, in run_app
    loop.run_until_complete(runner.setup())
  File "/home/ferran/.pyenv/versions/3.6.4/lib/python3.6/asyncio/base_events.py", line 467, in run_until_complete
    return future.result()
  File "/home/ferran/onna/onna-canonical/venv/lib/python3.6/site-packages/aiohttp/web_runner.py", line 172, in setup
    loop.add_signal_handler(signal.SIGINT, _raise_graceful_exit)
  File "/home/ferran/.pyenv/versions/3.6.4/lib/python3.6/asyncio/unix_events.py", line 93, in add_signal_handler
    raise RuntimeError(str(exc))
RuntimeError: set_wakeup_fd only works in main thread
```

2) We can start the worker before the sever, otherwise the server errors complaining that the loop is already running.

**Comment by lferran** [comment]
ready to merge?

**Comment by lferran** [comment]
@vangheem I think it's fine for now. If we ever find the need to actually spawn the complete guillotina server we can do it then.

**Pull request merged**